common/flatpak-dir.c

```diff
@@ -14359,10 +14363,10 @@ flatpak_dir_find_remote_related (FlatpakDir         *self,
                                    state->remote_name,
                                    &url,
                                    error))
-    return FALSE;
+    return NULL;
 
   if (*url == 0)
-    return g_steal_pointer (&related);  /* Empty url, silently disables updates */
+    return NULL;  /* Empty url, silently disables updates */
```

Comment:
Why is this change being made? It breaks with the convention of setting `error` if and only if a function returns `NULL` which returns `non-NULL` on success.

I also don't understand why we'd want the `*url == 0` case to have different behavior here vs in `flatpak_dir_find_remote_related_for_metadata()`

Reply:
To be fair [the GError rules](https://developer.gnome.org/glib/stable/glib-Error-Reporting.html#gerror-rules) say that `NULL` can be a non-error return value, but I don't think this is a good case for adding that complexity. And if we were to do that, we'd have to update the other call sites too like `flatpak_installation_list_remote_related_refs_sync()`.

Reply:
Yeah, this seems very wrong to me.

Reply:
> different behavior here vs in flatpak_dir_find_remote_related_for_metadata()

In the latter, the `related` was initialised with empty array and in the former it wasn't.
That's why the different behavior in the old diff.

I have updated the PR to bring those in-sync. Please take a look.